lib/groups.js

```diff
@@ -139,12 +179,54 @@ module.exports.createGroup = function (groupName, assessmentId, userId, authnUse
 };
 
 module.exports.leaveGroup = function (assessmentId, userId, authnUserId, callback) {
+  const params = [assessmentId, userId, authnUserId];
+  sqldb.call('group_leave', params, function (err) {
+    if (ERR(err, callback)) return;
+    callback();
+  });
+};
+
+module.exports.getAssessmentLevelPermissions = function (assessmentId, userId, callback) {
   const params = {
     assessment_id: assessmentId,
     user_id: userId,
-    authn_user_id: authnUserId,
   };
-  sqldb.query(sql.leave_group, params, function (err) {
+  sqldb.query(sql.get_assessment_level_permissions, params, function (err, result) {
+    if (ERR(err, callback)) return;
+    callback(result.rows[0]);
+  });
+};
+
+module.exports.updateGroupRoles = function (
+  requestBody,
+  assessmentId,
+  userId,
+  authnUserId,
+  callback
+) {
+  // Convert form data to valid input format for a SQL function
+  const roleKeys = Object.keys(requestBody).filter((key) => !key.startsWith('__'));
```

Comment:
Rather than looking specifically for things that *don't* start with `__`, I'd recommend adding a known prefix to the input names, e.g. `user_role_`.